client/settings/general-settings-section/payment-method-checkbox.js

```diff
@@ -44,9 +45,17 @@ const PaymentMethodCheckbox = ( {
 	] = useEnabledPaymentMethodIds();
 	const [ , setIsStripeEnabled ] = useIsStripeEnabled();
 	const { isUpeEnabled } = useContext( UpeToggleContext );
+	let checked;
+	if ( disabled ) {
+		checked = false;
+	} else if ( disabledButChecked ) {
+		checked = true;
+	} else {
+		checked = enabledPaymentMethods.includes( id );
+	}
```

Comment:
[nit] This is more of personal preference... feel free to ignore it!
```suggestion
	const checked = !disabled && ( disabledButChecked || enabledPaymentMethods.includes( id ) );
```